### DISCUSSION TITLE
Sending images in a modified `callback_query`

### DISCUSSION THREAD

**Discussion by olegkorol** [python-telegram-bot/python-telegram-bot#2259] [Q&A] [answered]
Hello everyone!

I am prototyping a small shop-bot built with the ConversationHandler and CallbackQueryHandler’s.
In order to provide smooth answers to the end-user, I use callback_query.edit_message_text().

At some given point I would also like to provide images, but I am struggling to do so.
I tried using callback_query.edit_message_media() but it doesn’t work since the message did not have any media previously.
Using context.bot.send_photo() does not seem to be right either, as it creates a completely new and separate message.

Here’s an example of the flow that I am trying to build:

Step 1
https://ibb.co/1XV7KYw
Step 2
https://ibb.co/cvkcFj9
Step 3
https://ibb.co/Cn4FpMV
Step 4
https://ibb.co/YBH5PqB

What would be the best approach to achieve this?

Thanks a lot in advance!

**Comment by Bibo-Joshi** [comment, marked as answer]
Hi. You can't change a text message to a media message, that's right. I suggest to simply delete the current text message and attach the keyboard to the new message containing the picture.

**Reply by olegkorol** [reply]
Alright, thanks!
